notifier/notifier.go

```diff
@@ -485,22 +485,15 @@ func (n *Manager) sendAll(alerts ...*Alert) bool {
 				continue
 			}
 			// We can't use the cached values from previous iteration.
-			v1Payload, v2Payload = nil, nil
+			v2Payload = nil
 		}
 
 		switch ams.cfg.APIVersion {
 		case config.AlertmanagerAPIVersionV1:
 			{
-				if v1Payload == nil {
-					v1Payload, err = json.Marshal(amAlerts)
-					if err != nil {
-						level.Error(n.logger).Log("msg", "Encoding alerts for Alertmanager API v1 failed", "err", err)
-						ams.mtx.RUnlock()
-						return false
-					}
-				}
-
-				payload = v1Payload
+				level.Error(n.logger).Log("msg", "Alertmanager API v1 is deprecated and not long supported")
```

Comment:
I think we can remove this block and let `default:` take care of it.

Reply:
Plus with the pre-validation during unmarshalling this shouldn't be reachable, and even if it was, we shouldn't be returing as others could be v2

Reply:
👍